corehq/apps/domain/views.py

```diff
@@ -1975,7 +1975,11 @@ def post(self, request, *args, **kwargs):
                                           name=new_domain.documentation_file_path)
 
             for application in new_domain.full_applications():
-                original_id = application.copied_from._id
+                # Note that application is a build. If the original app has a build then application.copied_from
+                # will be a build and application.copied_from.copy_of will be the original app ID, otherwise
+                # application.copied_from will be the original app. (FB 190587) See also self.published_apps()
```

Comment:
I vaguely remember a lot of this stuff so it's not surprising, but I'm not sure I can justify any of it. Other than that we keep track of the history of copying apps from the exchange, which makes sense. I think since you end up publishing a specific build of an app, that's what ends up getting pointed to, for better or for worse.